python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -395,14 +397,19 @@ def mergeAndFilterReferences(self, exposure, refCats, refWcs):
         # parents are not within the exposure boundaries.  Note
         # that within a single input refCat, the parents always
         # appear before the children.
-        mergedRefCat = lsst.afw.table.SourceCatalog(refCats[0].table)
+        mergedRefCat = None
         for refCat in refCats:
+            refCat = refCat.get()
+            if mergedRefCat is None:
+                mergedRefCat = lsst.afw.table.SourceCatalog(refCat.table)
             containedIds = {0}  # zero as a parent ID means "this is a parent"
             for record in refCat:
                 if expPolygon.contains(record.getCoord().getVector()) and record.getParent() in containedIds:
                     record.setFootprint(record.getFootprint().transform(refWcs, expWcs, expRegion))
                     mergedRefCat.append(record)
                     containedIds.add(record.getId())
+        if mergedRefCat is None:
```

Comment:
Question more than a comment - will we get here only if refCats is an empty list? Would it ever be the case where refCats have empty tables, in which case this might still have to be raised? Sure, it wasn't raised earlier, but just wondering.

Reply:
Ah, reading it I think you're right, if all records are empty we wouldn't raise. I'm never quite sure what happens with empty things in general, and the best to handle it in particular. From what I gathered at discussions following DM-29862, I'm not sure there really _is_ a standard way of dealing with such cases yet.

As you say, we never raised before anyway; this addition was just Jim figuring we might as well throw a clearer exception since we needed to change the way we handled `mergedRefCat`. I'm going to go ahead and merge - I'm sure we can live with postponing handling the case of empty-only records until after this weekly gets tagged :)

Thanks for the instantaneous review, Arun!

Reply:
If `refCats` was empty, then line 398 before would have thrown an `IndexError`. Now, we make it throw a `RuntimeError`. I'm betting this doesn't make any difference, unless the file calling this was catching specific exceptions in which case that would have to be changed as well. Just wanted to point that out here.